src/Silex/Provider/TwigServiceProvider.php

```diff
@@ -54,15 +54,16 @@ public function register(Container $app)
         $app['twig.number_format.thousands_separator'] = ',';
 
         $app['twig'] = function ($app) {
-            $app['twig.options'] = array_replace(
+            $options = array_replace(
                 [
                     'charset' => $app['charset'],
                     'debug' => $app['debug'],
                     'strict_variables' => $app['debug'],
-                ], $app['twig.options']
+                ],
+                $app['twig.options']
             );
 
-            $twig = $app['twig.environment_factory']($app);
+            $twig = $app['twig.environment_factory']($options);
```

Comment:
As you said, this is a BC break, so it cannot be merged without a BC layer. I don't think this is acceptable as I can see several scenarios where one would override this service.

Reply:
This was just a quick mockup to illustrate my point. I understand it's not mergeable hence my question in https://github.com/silexphp/Silex/issues/1577#issuecomment-368353476